### PR TITLE
Server controller unit tests

### PR DESCRIPTION
**Pull request by joannanosek**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -26,22 +26,22 @@ jobs:
...
+          name: gometalinter and unit tests
           command: |
             cd ~/inference-model-manager/server-controller
             mkdir -p /go/src/github.com/IntelAI/inference-model-manager/server-controller
             mv * /go/src/github.com/IntelAI/inference-model-manager/server-controller
             cd /go/src/github.com/IntelAI/inference-model-manager/server-controller
             dep ensure -v
             make install-linter
-            make lint
+            make test
```
Comment (pawel-pieniazek) [inline .circleci/config.yml]: does `make test` also run `make lint`?

File: server-controller/hooks_test.go [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+	"github.com/intel/crd-reconciler-for-kubernetes/pkg/resource"
+	"github.com/intel/crd-reconciler-for-kubernetes/pkg/states"
+
+	crv1 "github.com/IntelAI/inference-model-manager/server-controller/apis/cr/v1"
+	"os"
+	"strings"
+)
+
+type mockClient struct {
```
Comment (pawel-pieniazek) [inline server-controller/hooks_test.go]: maybe mockClient could be in separate file?

File: .circleci/config.yml (hunk 1 above)
Comment (joannanosek) [reply .circleci/config.yml]: yep, it's one target in Makefile

**Comment by joannanosek** [comment]
@pawel-pieniazek it's about 16%

File: server-controller/hooks_test.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
+package main
```
Comment (mareklevv) [inline server-controller/hooks_test.go]: Add Intel license header

File: server-controller/mock_client.go [hunk 4]
```diff
@@ -0,0 +1,61 @@
+package main
```
Comment (mareklevv) [inline server-controller/mock_client.go]: Add Intel license header

**Pull request merged**